### PR TITLE
Feat: Add unvetter bot

### PR DESCRIPTION
**Pull request by F4ever**

### CODE REVIEW

File: src/transport/msg_types/ping.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    'stakingModuleIds': [int]
+}, ignore_extra_keys=True)
+
+
+def to_check_sum_address(msg: dict):
+    msg['guardianAddress'] = Web3.to_checksum_address(msg['guardianAddress'])
+    return msg
+
+
+UnvetMessageSchema = Schema({
```
Comment (Amuhar) [inline src/transport/msg_types/ping.py:25]: Should the 'unvet' schema be described here?

File: src/transport/msg_types/unvet.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    'signature': SignatureSchema,
+    'stakingModuleId': int,
+    'operatorIds': list[int],
+    'vettedKeysByOperator': list[int],
+}, ignore_extra_keys=True)
+
+
+class UnvetMessage(TypedDict):
+    type: str
+    blockNumber: int
```
[the remark this answers is not in the document]
Comment (F4ever) [reply src/transport/msg_types/unvet.py:27]: blockHash is not required
https://hackmd.io/@lido/rJrTnEc2a#Summary-of-Interface-Changes

File: src/transport/msg_types/ping.py (hunk 1 above)
Comment (F4ever) [reply src/transport/msg_types/ping.py:25]: What do you mean describe?

File: src/bots/unvetter.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            filters=[
+                message_metrics_filter,
+                to_check_sum_address,
+                get_unvet_messages_sign_filter(unvet_prefix),
+            ],
+        )
+
+    def execute(self, block: BlockData) -> bool:
+        if not self.fully_initialized:
+            if self.w3.lido.deposit_security_module.__class__.__name__ == 'DepositSecurityModuleContractV2':
```
Comment (F4ever) [inline src/bots/unvetter.py:66]: Check unveting bot proper start and blockchain state in separate functions with comments

File: src/bots/unvetter.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        return message_filter
+
+    def _send_unvet_message(self, message: UnvetMessage) -> bool:
+        module_id = message['stakingModuleId']
+
+        logger.warning({'msg': f'Handle unvet message for module: {module_id}', 'value': message})
+
+        actual_nonce = self.w3.lido.staking_router.get_staking_module_nonce(module_id)
+
+        if message['nonce'] < actual_nonce:
```
Comment (F4ever) [inline src/bots/unvetter.py:111]: no need for this condition

File: src/depositor.py [hunk 5]
```diff
@@ -1,64 +1,22 @@
-# from flashbots import flashbot
 from eth_account import Account
```
Comment (F4ever) [inline src/depositor.py:1]: Make sure old dockercompose file works

File: src/transport/msg_types/unvet.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Amuhar) [reply src/transport/msg_types/unvet.py:27]: document is outdated

 {
    "inputs": [
      {
        "internalType": "uint256",
        "name": "blockNumber",
        "type": "uint256"
      },
      {
        "internalType": "bytes32",
        "name": "blockHash",
        "type": "bytes32"
      },
      {
        "internalType": "uint256",
        "name": "stakingModuleId",
        "type": "uint256"
      },
      {
        "internalType": "uint256",
        "name": "nonce",
        "type": "uint256"
      },
      {
        "internalType": "bytes",
        "name": "nodeOperatorIds",
        "type": "bytes"
      },
      {
        "internalType": "bytes",
        "name": "vettedSigningKeysCounts",
        "type": "bytes"
      },
      {
        "components": [
          {
            "internalType": "bytes32",
            "name": "r",
            "type": "bytes32"
          },
          {
            "internalType": "bytes32",
            "name": "vs",
            "type": "bytes32"
          }
        ],
        "internalType": "struct DepositSecurityModule.Signature",
        "name": "sig",
        "type": "tuple"
      }
    ],
    "name": "unvetSigningKeys",
    "outputs": [],
    "stateMutability": "nonpayable",
    "type": "function"
  }

File: src/transport/msg_types/ping.py (hunk 1 above)
Comment (Amuhar) [reply src/transport/msg_types/ping.py:25]: it is ping.py

File: src/transport/msg_types/pause.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    # 'stakingModuleId': int
+}, ignore_extra_keys=True)
+
+
+class PauseMessage(TypedDict):
+    type: str
+    blockNumber: int
+    guardianAddress: str
+    signature: Signature
+    stakingModuleId: int
```
Comment (Amuhar) [inline src/transport/msg_types/pause.py:45]: Should we remove stakingModuleId?

File: src/bots/unvetter.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        unvet_tx = self.w3.lido.deposit_security_module.unvet_signing_keys(
+            message['blockNumber'],
+            module_id,
+            message['nonce'],
+            message['operatorIds'],
+            message['vettedKeysByOperator'],
+            (message['signature']['r'], compute_vs(message['signature']['v'], message['signature']['s']))
+        )
+
+        result = self.w3.transaction.send(unvet_tx, False, 6)
```
[the remark this answers is not in the document]
Comment (F4ever) [reply src/bots/unvetter.py:123]: It works and for classic send also

File: src/transport/msg_types/pause.py (hunk 6 above)
Comment (F4ever) [reply src/transport/msg_types/pause.py:45]: It's required to support current version

File: src/transport/msg_types/ping.py (hunk 1 above)
Comment (F4ever) [reply src/transport/msg_types/ping.py:25]: Yeah! Thx

File: src/transport/msg_types/unvet.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (F4ever) [reply src/transport/msg_types/unvet.py:27]: Fixed

File: tests/bots/test_unvetter.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+def get_unvet_message(web3) -> UnvetMessage:
+    latest = web3.eth.get_block('latest')
+    block_number = latest.number
+
+    prefix = web3.lido.deposit_security_module.get_unvet_message_prefix()
+    nonce = web3.lido.staking_router.functions.getStakingModuleNonce(1).call()
+
+    msg_hash = web3.solidity_keccak(
+        ['bytes32', 'uint256', 'uint256', 'uint256[]', 'uint256[]'],
+        [prefix, block_number, nonce, [1], [1]],
```
Comment (Amuhar) [inline tests/bots/test_unvetter.py:25]: why dont you sign block hash and saking module id too ?

File: src/blockchain/web3_extentions/lido_contracts.py [hunk 9]
```diff
@@ -32,12 +45,25 @@ def _load_contracts(self):
...
+        dsm_address = self.lido_locator.deposit_security_module()
+
+        self.deposit_security_module = cast(DepositSecurityModuleContractV2, self.w3.eth.contract(
+            address=dsm_address,
+            ContractFactoryClass=DepositSecurityModuleContractV2,
+        ))
+
+        try:
+            self.deposit_security_module.version()
+        except ContractLogicError:
```
Comment (Amuhar) [inline src/blockchain/web3_extentions/lido_contracts.py:62]: If the issue wasn't caused by the absence of a method, but rather by another network error that resulted in a "ContractLogicError," how would we handle it?

File: tests/bots/test_unvetter.py (hunk 8 above)
Comment (F4ever) [reply tests/bots/test_unvetter.py]: fixed!

File: README.md [hunk 10]
```diff
@@ -10,6 +10,8 @@ This deposit is executed using the depositBufferedEther function within the "Dep
```
Comment (Amuhar) [inline README.md:10]: It seems like the test section should be changed.

File: src/transport/msg_types/pause.py [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+from transport.msg_types.base import ADDRESS_REGREX, SignatureSchema, Signature
+
+
+logger = logging.getLogger(__name__)
+
+
+"""
+Pause msg example:
+{
+    "blockHash": "0xe41c0212516a899c455203e833903c802338daa3048bc637b623f6fba0a1685c",
```
Comment (Amuhar) [inline src/transport/msg_types/pause.py:18]: Do we need blockHash?

**Review by infloop — APPROVED** [review]

**Pull request merged**